src/foam/nanos/notification/email/DAOEmailService.js

```diff
@@ -92,11 +92,23 @@ return config_;`
       name: 'sendEmail',
       javaCode: `
         getDao().put(emailMessage);
+`
+    },
+    {
+      name: 'sendEmailWithX',
```

Comment:
Why the double calls? Why not just fix everything? We have to update all calls to pass the context, that’s the entire point, is it not?

Reply:
Only for the backward compatibility. We have about 10~15 places are calling this method. I think it's better to change them progressively.